packages/amplify-codegen/src/utils/getCodegenPackageName.js

```diff
@@ -0,0 +1,12 @@
+const CODEGEN_OLD_PACKAGE = 'amplify-codegen-appsync-model-plugin';
+const CODEGEN_NEW_PACKAGE = '@aws-amplify/appsync-modelgen-plugin';
+
+function getCodegenPackageName(isNewCodegenPackage) {
```

Comment:
I think it would be better to require both the module and then send the right one based on the flag. If the packages are missing dynamic imports defer the problem till the specific code path is hit

Reply:
done

Reply:
Can you import this at  the top level. Otherwise there is no benefit doing this

Reply:
import moved to top and added a custom property "name" in the mockModule